### PR TITLE
oomd: only check prefix_uid when uid != 0

### PR DESCRIPTION
**Pull request by enr0n**
This is a follow-on to address https://github.com/systemd/systemd/pull/25509#discussion_r1031539489.

### CODE REVIEW

File: src/oom/oomd-util.c [hunk 1]
```diff
@@ -145,7 +145,7 @@ bool oomd_swap_free_below(const OomdSystemContext *ctx, int threshold_permyriad)
 }
 
 int oomd_fetch_cgroup_oom_preference(OomdCGroupContext *ctx, const char *prefix) {
-        uid_t uid, prefix_uid;
+        uid_t uid = UID_INVALID, prefix_uid = UID_INVALID;
```
Comment (yuwata) [inline src/oom/oomd-util.c]: It is not necessary to initialize `uid`, as `cg_get_owner()` always assign a value on success.

File: src/oom/oomd-util.c [hunk 2]
```diff
@@ -160,11 +160,13 @@ int oomd_fetch_cgroup_oom_preference(OomdCGroupContext *ctx, const char *prefix)
...
                 return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
 
-        r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, prefix, &prefix_uid);
-        if (r < 0)
-                return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
+        if (uid != 0) {
+                r = cg_get_owner(SYSTEMD_CGROUP_CONTROLLER, prefix, &prefix_uid);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to get owner/group from %s: %m", ctx->path);
+        }
```
Comment (yuwata) [inline src/oom/oomd-util.c:174]: Maybe, better to check `uid == prefix_uid` here, and set `MANAGED_OOM_PREFERENCE_NONE` if the condition does not match:
```c
if (uid != 0) {
    uid_t prefix_uid;

    r = cg_get_owner(..., &prefix_uid);
    if (r < 0)
        return log_debug_errno(r, "...");

    if (prefix_uid != uid) {
        ctx->preference = MANAGED_OOM_PREFERENCE_NONE;
        return 0;
    }
}

r = cg_get_xattr_bool(...);
...
```

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/oom/oomd-util.c (hunk 2 above)
Comment (enr0n) [reply src/oom/oomd-util.c:174]: Thanks, I agree this is better.

**Comment by enr0n** [comment]
I also updated relevant documentation in the last push.

**Review by yuwata — APPROVED** [review]

**Comment by poettering** [comment]
excellent! thnanks!

**Pull request merged**